test/reverse.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var throws = require('assert').throws;
+
+var eq = require('./utils').eq;
+var errorEq = require('./utils').errorEq;
+var S = require('..');
+
+
+describe('reverse', function() {
+
+  it('type checks its arguments', function() {
+    throws(function() { S.reverse({answer: 42}); },
+           errorEq(TypeError,
+                   'Invalid value\n' +
+                   '\n' +
+                   'reverse :: List a -> List a\n' +
+                   '           ^^^^^^\n' +
+                   '             1\n' +
+                   '\n' +
+                   '1)  {"answer": 42} :: Object, StrMap Number, StrMap FiniteNumber, StrMap NonZeroFiniteNumber, StrMap Integer, StrMap ValidNumber\n' +
+                   '\n' +
+                   'The value at position 1 is not a member of ‘List a’.\n'));
+
+  });
+
+  it('is a unary function', function() {
+    eq(typeof S.reverse, 'function');
+    eq(S.reverse.length, 1);
+  });
```

Comment:
I thought you just asked me to move the typecheck test to the top 😅 - anyway i will move it